pkg/cmd/render/render_test.go

```diff
@@ -488,3 +490,105 @@ type fakeBootstrapIPLocator struct {
 func (f *fakeBootstrapIPLocator) getBootstrapIP(ipv6 bool, machineCIDR string, excludedIPs []string) (net.IP, error) {
 	return f.ip, nil
 }
+
+const installConfigSingleStackIPv4 = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+const installConfigDualStack = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 2620:52:0:1302::/64
+  - cidr: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+const installConfigSingleStackIPv6 = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 2620:52:0:1302::/64
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+func Test_getMachineCIDR(t *testing.T) {
+	tests := map[string]struct {
+		installConfig     string
+		isSingleStackIPv6 bool
+		expectedCIDR      string
+		expectedErr       error
+	}{
+		"ipv4 should ignore the ipv6 cidr": {
+			installConfig:     installConfigSingleStackIPv4,
```

Comment:
description is bad; there is no ipv6 cidr to be ignored here

Reply:
d'oh, fixed (and also tried to improve all the descriptions)